Sources/Core/Video/VideoDecoding.swift

```diff
@@ -0,0 +1,85 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2015-2021 Alexander Grebenyuk (github.com/kean).
+
+
+#if !os(watchOS)
+
+import Foundation
+import AVKit
+
+extension ImageType {
+    public static let mp4: ImageType = "public.mp4"
+}
+
+extension ImageDecoders {
+    public final class Video: ImageDecoding, ImageDecoderRegistering {
+        private var didProducePreview = false
+
+        static func isVideo(_ data: Data) -> Bool {
+            match(data, offset: 4, [0x66, 0x74, 0x79, 0x70])
+        }
+
+        public var isAsynchronous: Bool {
+            false
```

Comment:
I think it should be asynchronous depending on how slow the creation of an asset is (assuming it will be moved here). Async means "executed on an operation queue". An alternative is executing on the image pipeline internal serial queue (slightly less overhead for quick operations).